### ISSUE TITLE
Storage Drawer Issues

### ISSUE THREAD

**Issue by MaxFirestorm** [NillerMedDild/Enigmatica6#225]
**Modpack version**
0.2.14

**Issue**
Issue 1:
Storage Drawers that are locked to an item but have none of that item inserted don't "remember" their locked item when placed from inventory or on world load. This causes the drawer to be locked but empty, not allowing any items into it at all. Expected behaviour is that the drawer remembers the item, regardless of if there is any of it present in the drawer at the time. This also seems to happen when the world is loaded, causing some issues with automation setups (for example locked drawer for wheat for a crusher for Mek Ethylene production.

Issue 2:
When Refined Storage External Storages are connected to Storage Drawer Drawer Controllers on world load, the External Storage doesn't read the contents of the connected drawers at all. My current example is that I have mob farm loot connected via drawer system but on world loading, the drawer system doesn't get read and instead dumps all input from the farm into the discs until I break/replace the External Storage. This is currently happening with other systems around my base as well.

**Comment by NillerMedDild** [comment]
Sounds like #173, are you using a Logistical Sorter?

**Comment by MaxFirestorm** [comment]
Nope, not using Mek tubes for this stuff at all. Purely Refined Storage.

**Comment by novaexe** [comment]
can confirm drawers also do this with hoppers and modular routers

**Comment by novaexe** [comment]
https://github.com/jaquadro/StorageDrawers/issues/863

**Comment by NillerMedDild** [comment]
I see, thanks for the link 👍 
Also: https://github.com/jaquadro/StorageDrawers/issues/810

**Comment by Shindur** [comment]
Having same issue regarding the external storage. I'll break and place the controller to get it working again, but it is quite irritating.

**Comment by Mendanbar** [comment]
I started seeing the second issue (drawer controller not reporting inventory to RS until connection is broken and reestablished) in 0.4.x.  But it could be because I've been restarting the server more since the update. I really wish RS would offer an easy void function, then I could ditch drawers altogether

**Comment by TemperedFool** [comment]
Better than a void option for almost all setups, you can use detectors to redstone-control things and turn them off when stocks are sufficient.

**Comment by Mendanbar** [comment]
> Better than a void option for almost all setups, you can use detectors to redstone-control things and turn them off when stocks are sufficient.

Yeah I've used this for cobble gen in the past, but the convenience of the void upgrade wins in situations where it's complicated to turn off a generation mechanism.

**Comment by NillerMedDild** [comment]
https://github.com/mekanism/Mekanism/issues/7023

**Comment by TemperedFool** [comment]
I get the convenience factor @Mendanbar - I've taken it myself plenty of times! - but it's bad form from a performance POV. Probably doesn't matter in SSP, but it's a good habit to build some sort of way to turn things off

**Comment by Mendanbar** [comment]
> Fingers crossed! [mekanism/Mekanism#7023](https://github.com/mekanism/Mekanism/issues/7023)

Didn't sound too positive.  The best use case for the Mek void upgrade is their own PRC IMO.  It pumps out tons of substrate that has a fairly narrow use case.  Also I'm super confused at their staunch opposition considering ALL of the Mek gas containers/machines have a "dump excess" function.  Seems like a silly place to dig your feet in.

At any rate I'll keep using drawers to void in situations like I listed above.  In the PRC case it doesn't seem like chute>lava is any more efficient. 🤷‍♂️

**Comment by NillerMedDild** [comment]
I believe the Gas containers have a dump excess button because it makes sense to be able to release gases into the atmosphere - Doing so with a radioactive gas will even cause the area to become irradiated.

**Comment by Mendanbar** [comment]
Yah I know, but we're talking about a mod that has teleportation as well... Not a far stretch to have a "void teleport" option for items. I know they have their own reasons, and that's fair. I just find this particular one silly.

**Comment by Mendanbar** [comment]
Also I'm ignoring the fact that some of the gases would def not be safe to vent at the levels I do in game (don't set off a flint and steel in my base lol)

**Comment by MuteTiefling** [comment]
We have trash cans (those also work on gases, BTW, but not radioactive ones).

They're more than ample for dealing with overflow items. Heck, I usually only save a little substrate and then straight up trash the trash.

**Comment by Mendanbar** [comment]
> We have trash cans (those also work on gases, BTW, but not radioactive ones).
> 
> They're more than ample for dealing with overflow items. Heck, I usually only save a little substrate and then straight up trash the trash.

Yep I've done the same in packs where drawers are not a thing.  Lots of options.  Chute->lava, pipe->trash, etc.  All of them take more setup than a drawer void upgrade, and (most important for me) make the build more messy.  I guess my real point is that Mek has a chance to address the issue directly.  But I guess they have philosophical reasons preventing them from doing so.  I think it would be a nice addition though, especially for ethylene substrate setups.

At any rate, I'm not arguing for anything in particular at this point.. just dreaming :). I tore down my PRC Ehtylene setup last night in favor of a Powah! reactor anyway lol.  I'm loving the changes to the Powah! reactors so far btw. 👍

**Comment by TemperedFool** [comment]
Priority on the sidedness of machines would help; then it could eject to a bin until its full, and then eject to a trash can (or dispenser->lava, or whatever). Problem for the mod though ^_^

**Comment by MuteTiefling** [comment]
> At any rate, I'm not arguing for anything in particular at this point.. just dreaming :). I tore down my PRC Ehtylene setup last night in favor of a Powah! reactor anyway lol. I'm loving the changes to the Powah! reactors so far btw. 👍

<3

**Comment by Wishbone1977** [comment]
Issue 2 has been known about for a long time, with no fix in sight. I have the same problem in my E6 world, but I've encountered it in my own pack as well. [https://github.com/refinedmods/refinedstorage/issues/2636](https://github.com/refinedmods/refinedstorage/issues/2636)
